netspeed/src/backend.c

```diff
@@ -105,6 +111,77 @@ get_available_devices (void)
     return device_glist;
 }
 
+GSList*
+get_ip_address_list (const char *ifa_name,
+                     gboolean    ipv4)
+{
+    GSList *list = NULL;
+    struct ifaddrs *ifaddr;
+
+    if (getifaddrs (&ifaddr) != -1) {
+        int family;
+        char ip[INET6_ADDRSTRLEN];
+
+        family = ipv4 ? AF_INET : AF_INET6;
+        for (struct ifaddrs *ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
+            if (ifa->ifa_addr == NULL)
+                continue;
+
+            if ((ifa->ifa_addr->sa_family == family) &&
+                !g_strcmp0 (ifa->ifa_name, ifa_name))
+            {
+                void *tempAddrPtr = NULL;
+
+                if (ifa->ifa_addr->sa_family == AF_INET6) {
+                    const char *scope;
+                    guint8 prefix6[16];
+                    guint  word, shiff;
+                    guint  prefix_lenght = 0;
+                    gboolean check = TRUE;
+
+                    tempAddrPtr = &((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_addr;
```

Comment:
Build warning.
```
backend.c:142:37: warning: cast from 'struct sockaddr *' to 'struct sockaddr_in6 *' increases required alignment from 2 to 4 [-Wcast-align]

                    tempAddrPtr = &((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_addr;

                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750455224#L5932

Reply:
Hmm, warnings are still there.
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750571886#L5929

Reply:
I can't see it on my local build:
```
$ CFLAGS="-Wcast-align" ./autogen.sh --prefix=/usr && make &> make.log
$ grep -F "Wcast-align" make.log 
```

Reply:
Confirmed, i don't get that warning when building local with  `-Wcast-align`, but in latest debian build from this branch it is there. https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750599329#L4410
Any way, it's not reproducible for me.